### PR TITLE
[PAY-287] Add prompt to become top or first supporter

### PR DESCRIPTION
**Pull request by sddioulde**
### Description
Add prompt to become top or first supporter in send tip screen.

Simulator Screen Shot - iPhone 13 - 2022-06-08 at 14 41 32

### Dragons

n/a

### How Has This Been Tested?

local mobile against local dapp against stage

### How will this change be monitored?

n/a

### CODE REVIEW

File: packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx [hunk 1]
```diff
@@ -50,12 +64,112 @@ const parseToBNWei = (tipAmount: StringAudio) => {
...
+    if (!account || !receiver) return
+    if (supportingAmount) return
+
+    const supportingForAccount = supportingMap[account.user_id] ?? null
+    const accountSupportingReceiver =
+      supportingForAccount?.[receiver.user_id] ?? null
+    if (accountSupportingReceiver) {
+      setSupportingAmount(accountSupportingReceiver.amount)
+    } else {
+      dispatchWeb({
```
Comment (rickyrombo) [inline packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx:104]: Maybe update web to dispatch that action too so both web and mobile do the same thing? https://github.com/AudiusProject/audius-client/pull/1415/files#diff-a18b6f1a60d7dfb83854657981e14cbab2fe2b5f9f0dbcbaee95e9d603770655R94

File: packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx [hunk 2]
```diff
@@ -50,12 +64,112 @@ const parseToBNWei = (tipAmount: StringAudio) => {
...
+
+  /**
+   * Get user who is top supporter to later check whether it is
+   * not the same as the current user
+   */
+  useEffect(() => {
+    if (!receiver) return
+
+    const supportersForReceiver = supportersMap[receiver.user_id] ?? {}
+    const rankedSupportersList = Object.keys(supportersForReceiver)
```
Comment (rickyrombo) [inline packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx:121]: Might be worth making a utility function to convert the map to a ranked list, as I think we do this in a few places now?

**Review by rickyrombo — APPROVED** [review]

File: packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx (hunk 2 above)
Comment (sddioulde) [reply packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx:121]: true that

File: packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx (hunk 1 above)
Comment (sddioulde) [reply packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx:104]: hmm interesting. yeah i like it

File: packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx [hunk 3]
```diff
@@ -50,12 +64,112 @@ const parseToBNWei = (tipAmount: StringAudio) => {
...
+
+  const tipAmountWei = parseToBNWei(tipAmount)
+  const hasInsufficientBalance = tipAmountWei.gt(accountBalance)
+
+  /**
+   * Get supporting info if current user is already supporting receiver
+   * so that the already supported amount can be used to determine
+   * how much is left to tip to become top supporter
+   */
+  useEffect(() => {
```
Comment (piazzatron) [inline packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx:94]: Discussed on slack consolidating this into hooks to share btwn desktop / web

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-top-first-supporter

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-top-first-supporter

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-top-first-supporter

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-top-first-supporter

**Pull request merged**